inject/maven-plugin/src/main/java/io/helidon/inject/maven/plugin/MavenOptions.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.inject.maven.plugin;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.helidon.codegen.CodegenOptions;
+
+class MavenOptions implements CodegenOptions {
+    private final Map<String, String> options;
+
+    private MavenOptions(Map<String, String> options) {
+        this.options = options;
+    }
+
+    public static CodegenOptions create(Set<String> compilerArgs) {
```

Comment:
there should be tests written for this.

Also, should we be doing validation of the options?  Especially if they have helidon in the names

Reply:
Validation of the option itself should be done while parsing it. Do you mean we should check if there is an unexpected option in our namespace?

Reply:
added validation, and fixed usages